synapse/cores/sqlite.py

```diff
@@ -510,6 +512,19 @@ def _getRowsByProp(self, prop, valu=None, limit=None, mintime=None, maxtime=None
         rows = self._runPropQuery('rowsbyprop',prop,valu=valu,limit=limit,mintime=mintime,maxtime=maxtime)
         return self._foldTypeCols(rows)
 
+    def _tufosByIn(self, prop, valus, limit=None):
+        ret = []
+
+        for valu in valus:
+            ret.extend(self.getTufosByProp(prop, valu=valu, limit=limit))
+
+            if limit != None:
+                limit -= len(ret)
```

Comment:
this is incorrect, you must assign getTufosByProp to some additional var and limit -= len(other)